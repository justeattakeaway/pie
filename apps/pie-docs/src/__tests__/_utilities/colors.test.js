const systemUnderTest = require('../../_utilities/colors');

describe('colors.js', () => {
    describe('isColorDark', () => {
        it.each([
            ['#000', true],
            ['#000000', true],
            ['#fff', false],
            ['#ffffff', false],
            ['#f36805', false]
        ])('hexcode %p returns %p', (tokenValue, expected) => {
            // act
            const result = systemUnderTest.isColorDark(tokenValue);

            // assert
            expect(result).toBe(expected);
        });
    });

    describe('convertHexcodeToRBG', () => {
        it.each([
            ['#f6c243'],
            ['#000'],
            ['#fff'],
            ['#000000'],
            ['#ffffff']
        ])('converts hexcode %p to the expected RGB object', (hexcode) => {
            // act
            const result = systemUnderTest.convertHexcodeToRBG(hexcode);

            // assert
            expect(result).toMatchSnapshot();
        });
    });
});
