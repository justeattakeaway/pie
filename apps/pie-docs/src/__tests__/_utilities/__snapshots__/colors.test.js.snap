// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #000 1`] = `
{
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;

exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #000000 1`] = `
{
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;

exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #f6c243 1`] = `
{
  "blue": 67,
  "green": 194,
  "red": 246,
}
`;

exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #fff 1`] = `
{
  "blue": 255,
  "green": 255,
  "red": 255,
}
`;

exports[`colors.js > convertHexcodeToRBG > converts hexcode %p to the expected RGB object #ffffff 1`] = `
{
  "blue": 255,
  "green": 255,
  "red": 255,
}
`;
